test/unit/luaT_tuple_new.c

```diff
@@ -187,6 +187,10 @@ main()
 	struct lua_State *L = luaL_newstate();
 	luaL_openlibs(L);
 
+	/* luaT_newmodule() assumes that this table exists. */
```

Comment:
Side note: I want to share here the approach used in [uJIT tests](https://github.com/luavela/luavela/blob/master/tests/impl/uJIT-tests-C/suite/test_common_lua.h#L27-L35), but I don't ask to fix it within this patchset. If this works for you, feel free to file an issue to move all unit tests for this machinery.

Reply:
Something like this?

```diff
diff --git a/test/unit/luaT_tuple_new.c b/test/unit/luaT_tuple_new.c
index 0b3847347..42643464c 100644
--- a/test/unit/luaT_tuple_new.c
+++ b/test/unit/luaT_tuple_new.c
@@ -14,6 +14,7 @@
 
 #define UNIT_TAP_COMPATIBLE 1
 #include "unit.h"             /* plan, header, footer, is, ok */
+#include "lua_test_utils.h"
 
 /*
  * This test checks all usage cases of luaT_tuple_new():
@@ -184,12 +185,7 @@ main()
 	memory_init();
 	fiber_init(fiber_c_invoke);
 
-	struct lua_State *L = luaL_newstate();
-	luaL_openlibs(L);
-
-	/* luaT_newmodule() assumes that this table exists. */
-	lua_newtable(L);
-	lua_setfield(L, LUA_REGISTRYINDEX, "_TARANTOOL_BUILTIN");
+	struct lua_State *L = luaT_newteststate();
 
 	box_init();
 	tarantool_lua_error_init(L);
diff --git a/test/unit/lua_msgpack.c b/test/unit/lua_msgpack.c
index 106097638..628ad1259 100644
--- a/test/unit/lua_msgpack.c
+++ b/test/unit/lua_msgpack.c
@@ -14,6 +14,7 @@
 
 #define UNIT_TAP_COMPATIBLE 1
 #include "unit.h"
+#include "lua_test_utils.h"
 
 static void
 mpstream_error_mock(void *ctx)
@@ -151,16 +152,11 @@ main(void)
 	plan(2);
 	header();
 
-	struct lua_State *L = luaL_newstate();
-	luaL_openlibs(L);
+	struct lua_State *L = luaT_newteststate();
 	tarantool_L = L;
 	memory_init();
 	fiber_init(fiber_c_invoke);
 
-	/* luaT_newmodule() assumes that this table exists. */
-	lua_newtable(L);
-	lua_setfield(L, LUA_REGISTRYINDEX, "_TARANTOOL_BUILTIN");
-
 	tarantool_lua_serializer_init(L);
 	luaopen_msgpack(L);
 	lua_pop(L, 1);
diff --git a/test/unit/lua_test_utils.h b/test/unit/lua_test_utils.h
new file mode 100644
index 000000000..1ed5dcb36
--- /dev/null
+++ b/test/unit/lua_test_utils.h
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#pragma once
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/**
+ * Incapsulates steps we should do to create a Lua state suitable
+ * to initialize some built-in Lua modules.
+ */
+static inline struct lua_State *
+luaT_newteststate(void)
+{
+	struct lua_State *L = luaL_newstate();
+	luaL_openlibs(L);
+
+	/* luaT_newmodule() assumes that this table exists. */
+	lua_newtable(L);
+	lua_setfield(L, LUA_REGISTRYINDEX, "_TARANTOOL_BUILTIN");
+
+	return L;
+}
+
+#if defined(__cplusplus)
+} /* extern "C" */
+#endif /* defined(__cplusplus) */
diff --git a/test/unit/lua_tweaks.c b/test/unit/lua_tweaks.c
index 56aa2b229..c5aa46253 100644
--- a/test/unit/lua_tweaks.c
+++ b/test/unit/lua_tweaks.c
@@ -15,6 +15,7 @@
 
 #define UNIT_TAP_COMPATIBLE 1
 #include "unit.h"
+#include "lua_test_utils.h"
 
 static struct lua_State *L;
 
@@ -239,13 +240,8 @@ main(void)
 {
 	memory_init();
 	fiber_init(fiber_c_invoke);
-	L = luaL_newstate();
-	luaL_openlibs(L);
-
-	/* luaT_newmodule() assumes that this table exists. */
-	lua_newtable(L);
-	lua_setfield(L, LUA_REGISTRYINDEX, "_TARANTOOL_BUILTIN");
 
+	L = luaT_newteststate();
 	tarantool_lua_tweaks_init(L);
 
 	/*
```

Added.

Reply:
Precisely, thanks!